### ISSUE TITLE
Some comments on workflow management dashboard

### ISSUE THREAD

**Issue by gaow** [bioworkflows/bioworkflows-help#18]
##  `Basic Info`

- It seems the way to update the workflow is to select `Clear any existing files and redownload` and we have to do that each time something worth updating happens on github?

## `Meta Info`

- `tag`: I think tags can be trick as it can be very arbitrary. You may end up having countless tags so I dont know if it is worth at all to have this option. I believe tags are meant to make searches in public workflows easier? Tags by definition is an attempt to sort. But if we believe in "search and not sort" perhaps we can instead use an optional section of "Description" that allows users to type more information. This is like an abstract to a paper and the search tool will parse text in the description and rank by relevance, for example. 
- `Reference`: It's a good idea. It is convenient to point to some online material eg a blog post link that part of the workflow is based on; but not so much a paper. If we can input the DOI number and get a bibtex citation in the background to convert to a reference entry that'd be great. Not sure if thye have an API that you can use: https://www.doi2bib.org/
- `License` : can we put in some default license types? So users can put in their names then choose a license. 

## `Accessibility`

- maintainer vs user: it's an interface thing, but I believe github allows one to add a name first then using a dropdown menu to change permissions eg edit or view only.
- Price vs token: Setting a price is most straightforward but I'm also thinking of setting up some tokens and tokens can be earned by either purchasing them, or contributing to bioworkflow.com. So the more active I am as a member contributing to the site the less or even nothing I have to pay for other people's work.

**Comment by BoPeng** [comment]
> It seems the way to update the workflow is to select Clear any existing files and redownload and we have to do that each time something worth updating happens on github?

If you do not select that and you have the same URL, the website does not know if your stuff has changed.   In the long run it will have github hooks ets so that your commits will trigger a hook in bioworkflows and get the workflow updated.

> tag: I think tags can be trick as it can be very arbitrary. You may end up having countless tags so I dont know if it is worth at all to have this option. I believe tags are meant to make searches in public workflows easier? Tags by definition is an attempt to sort. But if we believe in "search and not sort" perhaps we can instead use an optional section of "Description" that allows users to type more information. This is like an abstract to a paper and the search tool will parse text in the description and rank by relevance, for example.

Tags are designed to organize topics so that people can click 'gaow#rna` to get all `rna` tagged topics. It will also allow site-wide search (not ready yet). I agree with you that tags can be a mess though. 

The website also depends on a few specific tags such as `#tutorials` to categorize workflows.

I thought of description and gave it up because that is likely duplicated information from the main text and there is no appropriate place to show it. I plan to have some way to grab abstract from the main text, like right now the website is grabbing title if it was left unspecified.

> Reference: It's a good idea. It is convenient to point to some online material eg a blog post link that part of the workflow is based on; but not so much a paper. If we can input the DOI number and get a bibtex citation in the background to convert to a reference entry that'd be great. Not sure if thye have an API that you can use: https://www.doi2bib.org/

Great idea to have automatic doi and pubmed ID converted. I will create a separate ticket for it.

> License : can we put in some default license types? So users can put in their names then choose a license.

I will have a conference call with a "mentor" on license and IP related issues.

> Accessibility
maintainer vs user: it's an interface thing, but I believe github allows one to add a name first then using a dropdown menu to change permissions eg edit or view only.

Yes, right now these are two separate concept, one is to share (to a collaborator) for them to view, and one is to give users write permission to modify stuff. The current implementation does not allow more flexible ways though (e.g. convert viewer to maintainer) so perhaps I should have merged the two groups.

> Price vs token: Setting a price is most straightforward but I'm also thinking of setting up some tokens and tokens can be earned by either purchasing them, or contributing to bioworkflow.com. So the more active I am as a member contributing to the site the less or even nothing I have to pay for other people's work.

This goes along the line of "store credit" that can be earned and spent but not cash out. The problem with your suggestion is that I do not know how to define "activity", and appropriate "amount" that matches to the "quality", which are all ill-defined. Only money that people are willing to pay you shows real "money-worthing activity".

**Comment by gaow** [comment]
> The website also depends on a few specific tags such as #tutorials to categorize workflows.

Then I think tag should be predefined and chosen from. 

> duplicated information from the main text

It means the workflow themselves? If the search can be expanded to workflow text that should solve the problem. But maybe it makes more sense to have an `abstract` section in the workflow and search only that. Otherwise the search will be very messy too. That's why I thought a separate meta info with abstract can help making searches better.

**Comment by BoPeng** [comment]
> Then I think tag should be predefined and chosen from.

There is no way I can exhaust possible keywords for different types of research etc. perhaps I will end up dpomg something like "suggestions", meaning that existing tags from you and your groups will be suggested, followed by popular tags from the site, so there would be less "variations" of the same thing. 

I also plan to use tags to rate "tools" and "topics", so might introduce "prefixes" such as `tool#XXX`. When the site gets popular, there could be some statistics such as what tools have been used most often, what topics have been top researched etc. Basically all these are data that could be mined.

> It means the workflow themselves? 

Yes, imagine that you have a workflow page with the main text, and a side panel having the same abstract... 

Right now I am trying to simplify the creation process as much as possible so you essentially need two steps. All others are optional and useful only when you would like to share with others or publish. I think this is very important especially for people who use it only for archiving.

**Comment by BoPeng** [comment]
> Price vs token

I agree that we should encourage users to publish. Given that the publication process is moderated (require admin approval), there is not much a problem of spam. There are however several problems with tokens:

1. Whereas users would be happy to spend tokens, the authors might not be happy for receiving a bunch of tokens and be happier to receive real money, which then become a model for the website to pay a base fee for others to publish, which we cannot afford, at least for now.

2. Complications also arise for updated workflows, or derived workflows, which should earn less than the original version, and there is a slew of "value" related.

So there is a hard question on how to give appropriate amount of tokens for certain activities.
